services/geolocation/types.go

```diff
@@ -0,0 +1,46 @@
+package geolocation
+
+type GeoFetcher interface {
+	GeoIP(ip string) (*GeoCity, error)
+}
+
+// The City struct corresponds to the data in the GeoIP2/GeoLite2 City
+// databases. Given we are using the native library to decode the information,
+// we have modified some fields in it to contain the pointer values.
+type GeoCity struct {
```

Comment:
```suggestion
type GeolocationInfo struct {
```

Reply:
As per maxmind db, it is represented as GeoCity and hence the same naming convention. Let me know if you feel the name doesn't do justice ?